File: solutions/fsharp/raindrops/1/Raindrops.fs
module Raindrops

let rec divideOutFactors n factor =
    if n % factor = 0 then
        divideOutFactors (n / factor) factor
    else
        n

let rec convertRec (number: int) : string =
    match number with
    | n when n % 3 = 0 -> "Pling" + convertRec (divideOutFactors n 3)
    | n when n % 5 = 0 -> "Plang" + convertRec (divideOutFactors n 5)
    | n when n % 7 = 0 -> "Plong" + convertRec (divideOutFactors n 7)
    | _ -> ""

let convert (number: int) : string =
    let result = convertRec number
    if result = "" then number.ToString() else result

File: solutions/c/raindrops/1/raindrops.c
#include "raindrops.h"

#include <stdio.h>
#include <string.h>

static const int NUM_DROPS = 3;
static const int factors[] = {3, 5, 7};
static const char* drop_names[] = {"Pling", "Plang", "Plong"};

char* convert(char result[], int drops) {
    for (int i = 0; i < NUM_DROPS; i++) {
        if (drops % factors[i] == 0) strcat(result, drop_names[i]);
    }

    if (result[0] == '\0') sprintf(result, "%d", drops);
    return result;
}